enclave/core/sgx/calls.c

```diff
@@ -135,12 +136,58 @@ extern bool oe_disable_debug_malloc_check;
 **==============================================================================
 */
 
+#ifdef OE_WITH_EXPERIMENTAL_EEID
+extern oe_eeid_t* oe_eeid;
+extern const oe_enclave_config_t* oe_enclave_config;
+
+static oe_result_t _load_config(const oe_enclave_config_t* config)
+{
+    if (config && config->data && config->size)
+    {
+        if (oe_is_within_enclave(config->data, config->size) ||
+            oe_eeid->version != OE_EEID_VERSION)
+            return OE_INVALID_PARAMETER;
+
+        /* Copy config into enclave heap */
+        oe_enclave_config_t* copy = oe_malloc(sizeof(oe_enclave_config_t));
+        if (!copy)
+            return OE_OUT_OF_MEMORY;
+        copy->size = config->size;
+        copy->data = oe_malloc(config->size);
+        if (!copy->data)
+            return OE_OUT_OF_MEMORY;
+        memcpy(copy->data, config->data, config->size);
```

Comment:
This is the problematic line since `config` is under host control.

Reply:
This would allow host to control both `config->data` and `config->size` so that he can easily overwrite copy->data and trigger heap overflow.

Reply:
The host is in full control of `config` in all situations and at all times anyways, I don't think there's a problem with that here. I'll make a copy of `config` before running any checks, but the host can still pass in a `config->size` that's larger then the enclave memory space, but they might as well just refuse to load the enclave in the first place.

Reply:
I rewrote that bit now to take a copy of the host-provided values, so the size can't change anymore, but of course the host is still in control of the contents of `config->data`.